npm-cli/src/Pesy.re

```diff
@@ -36,188 +24,79 @@ let substituteTemplateValues = s =>
-    Path.(projectPath / "azure-pipelines.yml"),
-  );
-  mkdirp(".ci");
-  let ciFilesPath = Path.(projectPath / ".ci");
-  write(Path.(ciFilesPath / "esy-build-steps.yml"), esyBuildSteps);
-  copyTemplate(
-    Path.("azure-pipeline-templates" / "pesy-publish-build-cache.yml"),
-    Path.(ciFilesPath / "publish-build-cache.yml"),
-  );
-  copyTemplate(
-    Path.("azure-pipeline-templates" / "pesy-restore-build-cache.yml"),
-    Path.(ciFilesPath / "restore-build-cache.yml"),
-  );
-};
-
-let libKebab = packageNameKebabSansScope;
-let duneProjectFile = Path.(projectPath / "dune-project");
-if (!exists(duneProjectFile)) {
-  write(duneProjectFile, spf({|(lang dune 1.2)
-(name %s)
-|}, libKebab));
-};
-
-let opamFileName = libKebab ++ ".opam";
-let opamFile = Path.(projectPath / opamFileName);
-if (!exists(opamFile)) {
-  write(opamFile, "");
-};
-
-let rootDuneFile = Path.(projectPath / "dune");
-
-if (!exists(rootDuneFile)) {
-  write(rootDuneFile, "(ignored_subdirs (node_modules))");
-};
-
-/* TODO: Box drawing characters are transform to weird char by bucklescript. Investigate */
-/* Printf.printf("\x1b[1m%s@%s\x1b[0m\n", packageNameKebabSansScope, version); */
-/* Utils.renderAscTree([ */
-/*   [ */
-/*     "executable", */
-/*     spf( */
-/*       "bin:    %sApp.re as %sApp.exe", */
-/*       packageNameUpperCamelCase, */
-/*       packageNameUpperCamelCase, */
-/*     ), */
-/*     spf( */
-/*       "require: [%s]", */
-/*       List.fold_left( */
-/*         (acc, e) => acc ++ " " ++ e, */
-/*         "", */
-/*         [spf("\"%s\"", packageLibName)], */
-/*       ), */
-/*     ), */
-/*   ], */
-/*   ["library", "require: []"], */
-/*   [ */
-/*     "test", */
-/*     spf( */
-/*       "bin:    Test%s.re as Test%s.exe", */
-/*       packageNameUpperCamelCase, */
-/*       packageNameUpperCamelCase, */
-/*     ), */
-/*     spf( */
-/*       "require: [%s]", */
-/*       String.concat(", ", [spf("\"%s\"", packageLibName)]), */
-/*     ), */
-/*   ], */
-/* ]); */
-/* print_newline(); */
+download_git("github:ulrikstrid/hello-reason-pesy", projectPath, error =>
+  switch (error) {
+  | Some(e) => Js.log(e)
+  | None =>
+    let files =
+      walk_sync(projectPath)
+      ->Belt.Array.keep(file_or_dir => statSync(file_or_dir)->isFile);
+
+    Belt.Array.forEach(
+      files,
+      file => {
+        let () = readFile(file)->substituteTemplateValues |> write(file);
+
+        renameSync(file, substituteTemplateValues(file));
+      },
+    );
 
-let id = Spinner.start("\x1b[2mRunning\x1b[0m esy install");
-exec("esy i", (e, stdout, stderr) => {
-  Spinner.stop(id);
-  if (Js.eqNullable(e, Js.Nullable.null)) {
-    let id = Spinner.start("\x1b[2mRunning\x1b[0m esy pesy\x1b[2m and \x1b[0m building project dependencies");
-    exec("esy pesy", (e, stdout, stderr) => {
+    let libKebab = packageNameKebabSansScope;
+    let duneProjectFile = Path.(projectPath / "dune-project");
```

Comment:
I decided to keep generating a dune-project file and the .opam file, what do you think?